src/Symfony/Component/Security/Core/Security.php

```diff
@@ -12,27 +12,34 @@
 namespace Symfony\Component\Security\Core;
 
 use Psr\Container\ContainerInterface;
+use Symfony\Bundle\SecurityBundle\Security\Security as NewSecurityHelper;
 use Symfony\Component\Security\Core\Authentication\Token\TokenInterface;
 use Symfony\Component\Security\Core\Authorization\AuthorizationCheckerInterface;
 use Symfony\Component\Security\Core\User\UserInterface;
 
 /**
  * Helper class for commonly-needed security tasks.
  *
- * @final
+ * @deprecated since Symfony 6.2, use {@see \Symfony\Bundle\SecurityBundle\Security\Security} instead
  */
 class Security implements AuthorizationCheckerInterface
 {
+    // @deprecated In 7.0, move these constants to the NewSecurityHelper class and
+    // make them reference SecurityContextAttributes:* constants.
     public const ACCESS_DENIED_ERROR = '_security.403_error';
     public const AUTHENTICATION_ERROR = '_security.last_error';
     public const LAST_USERNAME = '_security.last_username';
     public const MAX_USERNAME_LENGTH = 4096;
```

Comment:
We should mark all constants using `/** @deprecated ... */`. At least Psalm is able to catch these.